staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -214,11 +215,21 @@ func (config *DirectClientConfig) getUserIdentificationPartialConfig(configAuthI
 	if len(configAuthInfo.Token) > 0 {
 		mergedConfig.BearerToken = configAuthInfo.Token
 	} else if len(configAuthInfo.TokenFile) > 0 {
-		tokenBytes, err := ioutil.ReadFile(configAuthInfo.TokenFile)
+		tokenFile, err := os.Open(configAuthInfo.TokenFile)
 		if err != nil {
 			return nil, err
 		}
-		mergedConfig.BearerToken = string(tokenBytes)
+		defer tokenFile.Close()
+
+		// read the first line (excluding newline) of the token file as
+		// the Bearer token
+		scanner := bufio.NewScanner(tokenFile)
```

Comment:
Make a bufio.Reader and call ReadLine or ReadString("\n") instead of using a Scanner?

Reply:
I can certainly make that change, however for this edit I looked through the codebase to get a sense of the right approach:

There are many uses of Scanner in the codebase, but fewer to just read a single line. Of the two alternatives, ReadString seems to be the better approach (ReadLine would need a continually filled buffer, which seems to be reimplementing logic already in ReadString and Scanner), but still needs a second step to strip the trailing newline.

When I made this change, my thinking was that Scanner is the better option as it encapsulates all the logic needed, doesn't require additional changes to the returned string, and is already in use in the codebase to read a single line more commonly than ReadString.

I'm happy to make the change if there's a compelling reason I don't know about as to why ReadString is the better approach, and I can use this as a learning opportunity.

Reply:
@lavalamp if you give me the nod that changing the code to use ReadString is a requirement for the PR to be accepted, I'm happy to make that change